lib/modules/navigationMiddleware.js

```diff
@@ -6,16 +6,30 @@ export function matchRoute(path, routes) {
   let route;
 
   for (route of routes) {
-    const [url, handler] = route;
+    const [url, handler, meta] = route;
     const reg = pathToRegex(url);
     const result = reg.exec(path);
 
     if (result) {
-      return { handler, reg, result };
+      return { handler, reg, result, meta };
     }
   }
 }
 
+const getRouteMeta = (routes, shouldSetPage, data) => {
+  const { method, pathName } = data;
+  const route = matchRoute(pathName, routes);
+
+  if (route && route.meta) {
+    const { meta } = route;
+
+    if (shouldSetPage && method === METHODS.GET) {
+      return meta;
+    }
+  }
+  return undefined;
```

Comment:
I think we may be diving too deep into semantics here. If someone told me a function called `getThing` would return 'thing' if it was valid, I wouldn't really bat an eye between `undefined` and `null` in the 'invalid' case. Returning `undefined` is the default behavior, so in someways I think we could easily make the argument for removing the return altogether.